### ISSUE TITLE
Can't publish preprocessor packages due to the transitive Neotoma dependency

### ISSUE THREAD

**Issue by pepicrft** [still-ex/still#186]
First of all, thanks for this great project. I'm new to the Elixir ecosystem and Still was one of the first tools I played with. I've implemented [a pre-processor to support .scss style files](https://github.com/craftweg/still_scss), but I can't push it to the Hex package registry due to the following error:

```bash
** (Mix) Can't build package with overridden dependency neotoma, remove `override: true`
```

Which happens because I had include the `neotoma` dependency in the `deps` function of my project to force Mix to use `rebar3`:

```exs
{:neotoma, "~> 1.7", manager: :rebar3, override: true}
```
Removing it leads to this other error due to dependency incompatibilities:

```bash
Unchecked dependencies for environment dev:
* neotoma (Hex package)
  the dependency neotoma in mix.exs is overriding a child dependency:

  > In mix.exs:
    {:neotoma, "~> 1.7", [env: :prod, repo: "hexpm", hex: "neotoma"]}

  > In deps/slime/mix.exs:
    {:neotoma, "~> 1.7", [env: :prod, hex: "neotoma", repo: "hexpm", optional: false]}

  Ensure they match or specify one of the above in your deps and set "override: true"
```

Am I right understanding that a solution to this problem comes down to configuring the `neotoma` dependency to build with `rebar3`?

Thanks in advance

**Comment by gabrielpoca** [comment]
@pepicrft yeah, I'm having the same issue with the latest versions of elixir/erlang. I still need to debug and fix this issue, but you can use the same version we are using to develop:

elixir 1.11.3-otp-23
erlang 23.2.3

Those should work. I'll keep you updated here.

**Comment by pepicrft** [comment]
@gabrielpoca thanks a lot for your suggestion. I'll go ahead with it and subscribe to this issue.

**Comment by pepicrft** [comment]
I tried with the suggested versions. This is what I get when I run `elixir --version`:

```
Erlang/OTP 23 [erts-11.1.7] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1]
Elixir 1.11.3 (compiled with Erlang/OTP 23)
```

However, I'm still getting the error:

```
Unchecked dependencies for environment dev:
* neotoma (Hex package)
  the dependency neotoma in mix.exs is overriding a child dependency:

  > In mix.exs:
    {:neotoma, "~> 1.7", [env: :prod, repo: "hexpm", hex: "neotoma"]}

  > In deps/slime/mix.exs:
    {:neotoma, "~> 1.7", [env: :prod, hex: "neotoma", repo: "hexpm", optional: false]}

  Ensure they match or specify one of the above in your deps and set "override: true"
** (Mix) Can't continue due to errors on dependencies
```

Am I missing anything?
Thanks @gabrielpoca

**Comment by gabrielpoca** [comment]
@pepicrft I was looking into it, and it seems there's a fix for it https://github.com/slime-lang/slime/issues/170#issuecomment-1136325884 can you try it out? Just adding `{:neotoma, "~> 1.7.3", manager: :rebar3, override: true}` was enough to make the error disappear on my system

**Comment by pepicrft** [comment]
:/ No luck. I get the following error:

```
** (Mix) Can't build package with overridden dependency neotoma, remove `override: true`
```

**Comment by gabrielpoca** [comment]
@pepicrft I'm sorry, l lost the notification. I was trying to understand how slime and phoenix_slime were publishing packages since they have the same dependency, but it seems that those packages haven't been updated for a long time. They will have the same issue once they release a new version. I also found your [comment in the forum](https://elixirforum.com/t/neotoma-is-using-rebar-2-error/47920/13?u=gabrielpoca), but no answer 😞 For now, I think you should keep it in Github, you can use it directly in your mix.exs. I hope this weird issue is not preventing you from investing in Elixir more 💪

**Comment by pepicrft** [comment]
Hey @gabrielpoca 
Thanks a lot for your answer. I appreciate a lot that you took the time to figure out a way to overcome the issue. I doubt this hiccup will impact my diving into the language. I'm liking it a lot indeed.
